### ISSUE TITLE
loss function

### ISSUE THREAD

**Issue by ejawany95** [Junzhou-Chen/EADB-Net#2]
First, thank you for sharing your work. 
I have a question regarding the loss calculation when n_classes == 1.

if model.n_classes == 1:
                        losses = [criterion(i(1), true_masks.float()) for i in masks_pred]
                        loss = sum(losses)
                        # loss = criterion(masks_pred.squeeze(1), true_masks.float())
                        losses = [dice_loss(F.sigmoid(i.squeeze(1)), true_masks.float(), multiclass=False) for i in
                                  masks_pred]
                        loss = sum(losses)

It seems that for n_classes == 1, the edge predictions (edge_pred) are not included in the loss computation. Is this intentional? If so, could you clarify why the edge predictions are omitted?

Additionally, I noticed that the last line only calculates the Dice loss for masks_pred. Shouldn't the final loss be a combination of both BCE and Dice loss (e.g., loss = bce_loss + dice_loss)?

I’d appreciate any clarification or corrections if I’ve misunderstood something.
